docs/reference/modules/admin-bar.md

```diff
@@ -0,0 +1,125 @@
+
+### `groups`
+
+The `groups` option takes an array of one or more objects that group several menu items together in the admin bar as a dropdown menu. Each of the `groups` objects requires a `label` and an array of menu `items`. The `label` will be used as the label displayed in the menu. The `items` array contains the names of the individual menu items you want to appear in the dropdown, entered in the order you want them to appear. Note: Menu names for `piece-type` items are the name of the piece-type, not the label. For core items, like 'Images', the name is prefixed - `@apostrophecms/image`.
+
+**Example**
+```javascript
+// modules/@apostrophecms/admin-bar/index.js
+module.exports = {
+ options: {
+   groups: [
+     {
+       name: 'media',
+       label: 'Media',
+       items: [
+         '@apostrophecms/image',
+         '@apostrophecms/file',
+         '@apostrophecms/image-tag',
+         '@apostrophecms/file-tag'
+       ]
+     }
+   ]
+ }
+};
+```
+This will result in grouping those four core modules into a single dropdown menu displayed as 'Media' on the menu bar.
+![ApostropheCMS admin bar with open dropdown menu titled 'Media'](/images/group-menu.png)
+
+## Featured methods
+
+The following method belongs to this module and may be useful in project-level code. See the [source code](https://github.com/apostrophecms/apostrophe/blob/main/modules/%40apostrophecms/admin-bar/index.js) for all methods that belong to this module.
+<!-- Some are used within the module and would just create noise here. -->
+
+Because this module has an alias, you can call these from another module from the alias path. For example, `self.apos.adminBar.add()`.
+
+### `add(name, label, permission, options)`
+
+Add an item to the menu bar.
+
+The `name` for the item must be unique within the menu bar to avoid conflicts. When the menu item is clicked, the `name` argument will be emitted on `apos.bus` as the value of an `admin-menu-click` event. If this item controls a specific modal, this will be caught by `TheAposModals` to display the correct modal. So, `name` should be the module name with a `:editor` or `:manager` suffix. For example, `@apostrophecms/global:editor`.
+
+**Example**
+```javascript
+apos.bus.$on('admin-menu-click', async (item) => {
+ // Make sure it is the button we care about, leave others to their own handlers
+ if (item !== 'myCustomWidget:editor') {
+   return;
+ }
+ // Custom code for button action
+});
+```
+The `label` will be the button name displayed on the menu bar.
+
+`permission` is optional and takes an object with `action` and `type` properties. If no permissions are present, anyone can perform the action represented by the button. The `action` property dictates what type of action the button will perform. These include `view`, `view-draft`, `edit`, `publish`, `upload-attachment`, and `delete`. The `type` property matches the name of the module the button is managing. This type must have a registered manager.
```

Comment:
Actual enforcement of permissions takes place elsewhere in the APIs that the button might invoke, so this is really just about who should see it: "If no permissions are present, anyone can see the button." The rest is great as written.